### ISSUE TITLE
`bosh clean-up --all` fails to delete job after upgrading release

### ISSUE THREAD

**Issue by cdutra** [cloudfoundry/bosh#2134]
**Describe the bug**
`bosh clean-up --all` fails to delete `loggr-forwarder-agent` job after upgrading `loggregator-agent-release` from `3.4` to `3.5`.
```
Task 148 | 22:25:02 | Deleting releases: cflinuxfs3/0.52.0 (00:00:00)
Task 148 | 22:25:02 | Deleting jobs: loggr-forwarder-agent/1b6449dad4a4ea3b7a2c7d5413ef7d763cdce74c92347b27aaeb72b8260dfc62 (00:00:00)
                    L Error: PG::ForeignKeyViolation: ERROR:  update or delete on table "templates" violates foreign key constraint "instances_templates_template_id_fkey" on table "instances_templates"
DETAIL:  Key (id)=(139) is still referenced from table "instances_templates".

Task 148 | 22:25:02 | Deleting releases: loggregator-agent/3.4 (00:00:00)
                    L Error: PG::ForeignKeyViolation: ERROR:  update or delete on table "templates" violates foreign key constraint "instances_templates_template_id_fkey" on table "instances_templates"
DETAIL:  Key (id)=(139) is still referenced from table "instances_templates".

Task 148 | 22:25:02 | Deleting packages: bbs/4ea26cdf576820d5409b1a6489619749c5b89c11 (00:00:00)
Task 148 | 22:25:02 | Deleting packages: golang-1-linux/c921dc1e1831da5a2dc61cb8945e372c1410c05882ed00965cba367991495f5d (00:00:00)
Task 148 | 22:25:02 | Deleting packages: golang-1-windows/378e0efca2c3c300004cdf78e82ab657894e43a18da3fac0e069ee7e7bf16267 (00:00:01)
Task 148 | 22:25:03 | Deleting packages: route_emitter/b051353e8de4dbfe79700a8c54b9f6d2f2a8c382 (00:00:00)
Task 148 | 22:25:03 | Deleting packages: route_emitter_windows/bab4d426e6c4e447f2f7eb859f5b6d8dcd753247 (00:00:00)
```

**To Reproduce**
```
bbl plan --lb-type=cf --lb-cert=./certs/fake.crt --lb-key=./certs/fake.key --lb-domain=domain.example
eval "$(bbl print-env)"
CF_D=/Users/pivotal/workspace/cf-deployment
pushd $CF_D
  git co fd773f6b
popd

bosh -d cf deploy -n \
  $CF_D/cf-deployment.yml \
  -o $CF_D/operations/use-compiled-releases.yml \
  -o $CF_D/operations/scale-database-cluster.yml \
  -o $CF_D/operations/stop-skipping-tls-validation.yml \
  -o $CF_D/operations/experimental/add-deployment-updater.yml \
  -o $CF_D/operations/experimental/enable-iptables-logger.yml \
  -o $CF_D/operations/experimental/set-cpu-weight.yml \
  -o $CF_D/operations/experimental/deploy-forwarder-agent.yml \
  -o $CF_D/operations/experimental/use-logcache-for-cloud-controller-app-stats.yml \
  -o $CF_D/operations/experimental/fast-deploy-with-downtime-and-danger.yml \
  -v system_domain=domain.example

bosh -d cf run-errand smoke-tests

pushd $CF_D
  git co d64e140d
popd

bosh -d cf deploy -n \
  $CF_D/cf-deployment.yml \
  -o $CF_D/operations/use-compiled-releases.yml \
  -o $CF_D/operations/scale-database-cluster.yml \
  -o $CF_D/operations/stop-skipping-tls-validation.yml \
  -o $CF_D/operations/experimental/add-deployment-updater.yml \
  -o $CF_D/operations/experimental/enable-iptables-logger.yml \
  -o $CF_D/operations/experimental/set-cpu-weight.yml \
  -o $CF_D/operations/experimental/deploy-forwarder-agent.yml \
  -o $CF_D/operations/experimental/use-logcache-for-cloud-controller-app-stats.yml \
  -o $CF_D/operations/experimental/fast-deploy-with-downtime-and-danger.yml \
  -v system_domain=domain.example

bosh clean-up --all # this fails
```

**Expected behavior**
`bosh clean-up --all` should successfully delete all the unused resources.

**Logs**
```
D, [2019-02-07T22:25:02.849448 #26010] [] DEBUG -- DirectorJobRunner: (0.001668s) (conn: 47388449880920) DELETE FROM "release_versions_templates" WHERE ("template_id" = 139)
D, [2019-02-07T22:25:02.849830 #26010] [] DEBUG -- DirectorJobRunner: (0.000035s) (conn: 47388435164720) BEGIN
D, [2019-02-07T22:25:02.851189 #26010] [] DEBUG -- DirectorJobRunner: (0.002003s) (conn: 47388448342640) DELETE FROM "release_versions_templates" WHERE (("template_id" = 128) AND ("release_version_id" = 45))
I, [2019-02-07T22:25:02.851414 #26010] []  INFO -- DirectorJobRunner: Keeping job cflinuxfs3-smoke-test/d4469ae07f84bcc0a3629618e08ed9ed26588afd as it is used by other release versions
D, [2019-02-07T22:25:02.853911 #26010] [] DEBUG -- DirectorJobRunner: (0.001826s) (conn: 47388435421440) DELETE FROM "release_versions_templates" WHERE (("template_id" = 37) AND ("release_version_id" = 45))
I, [2019-02-07T22:25:02.854065 #26010] []  INFO -- DirectorJobRunner: Remove all deployments in release version
D, [2019-02-07T22:25:02.854794 #26010] [] DEBUG -- DirectorJobRunner: (0.000185s) (conn: 47388449880920) DELETE FROM "deployments_release_versions" WHERE ("release_version_id" = 45)
D, [2019-02-07T22:25:02.855152 #26010] [] DEBUG -- DirectorJobRunner: (0.000039s) (conn: 47388448342640) BEGIN
D, [2019-02-07T22:25:02.856068 #26010] [] DEBUG -- DirectorJobRunner: (0.000769s) (conn: 47388448342640) DELETE FROM "release_versions" WHERE "id" = 45
D, [2019-02-07T22:25:02.858235 #26010] [] DEBUG -- DirectorJobRunner: (0.002026s) (conn: 47388448342640) COMMIT
D, [2019-02-07T22:25:02.859040 #26010] [] DEBUG -- DirectorJobRunner: (0.000258s) (conn: 47388435421440) SELECT * FROM "release_versions" WHERE ("release_versions"."release_id" = 9)
E, [2019-02-07T22:25:02.859668 #26010] [] ERROR -- DirectorJobRunner: PG::ForeignKeyViolation: ERROR:  update or delete on table "templates" violates foreign key constraint "instances_templates_template_id_fkey" on table "instances_templates"
DETAIL:  Key (id)=(139) is still referenced from table "instances_templates".: (conn: 47388435164720) DELETE FROM "templates" WHERE "id" = 139
D, [2019-02-07T22:25:02.860462 #26010] [] DEBUG -- DirectorJobRunner: (0.000048s) (conn: 47388449880920) BEGIN
D, [2019-02-07T22:25:02.861133 #26010] [] DEBUG -- DirectorJobRunner: (0.000276s) (conn: 47388449880920) UPDATE "tasks" SET "event_output" = ("event_output" || '{"time":1549578302,"stage":"Deleting releases","tags":[],"total":4,"task":"cflinuxfs3/0.52.0","index":1,"state":"finished","progress":100}
') WHERE ("id" = 148)
D, [2019-02-07T22:25:02.861408 #26010] [] DEBUG -- DirectorJobRunner: (0.001305s) (conn: 47388435164720) ROLLBACK
D, [2019-02-07T22:25:02.863207 #26010] [] DEBUG -- DirectorJobRunner: (0.001978s) (conn: 47388449880920) COMMIT
D, [2019-02-07T22:25:02.865004 #26010] [] DEBUG -- DirectorJobRunner: (0.000038s) (conn: 47388448342640) BEGIN
D, [2019-02-07T22:25:02.865794 #26010] [] DEBUG -- DirectorJobRunner: (0.000383s) (conn: 47388448342640) UPDATE "tasks" SET "event_output" = ("event_output" || '{"time":1549578302,"stage":"Deleting jobs","tags":[],"total":2,"task":"loggr-forwarder-agent/1b6449dad4a4ea3b7a2c7d5413ef7d763cdce74c92347b27aaeb72b8260dfc62","index":1,"state":"failed","progress":100,"data":{"error":"PG::ForeignKeyViolation: ERROR:  update or delete on table \"templates\" violates foreign key constraint \"instances_templates_template_id_fkey\" on table \"instances_templates\"\nDETAIL:  Key (id)=(139) is still referenced from table \"instances_templates\".\n"}}
') WHERE ("id" = 148)
D, [2019-02-07T22:25:02.866541 #26010] [] DEBUG -- DirectorJobRunner: (0.002322s) (conn: 47388449880920) DELETE FROM "locks" WHERE (("name" = 'lock:release:cflinuxfs3') AND ("uid" = '144afd84-5ebe-4460-9695-e317e45bbee7'))
D, [2019-02-07T22:25:02.866630 #26010] [] DEBUG -- DirectorJobRunner: Deleted lock: lock:release:cflinuxfs3 uid: 144afd84-5ebe-4460-9695-e317e45bbee7
D, [2019-02-07T22:25:02.868131 #26010] [] DEBUG -- DirectorJobRunner: (0.000120s) (conn: 47388435421440) BEGIN
D, [2019-02-07T22:25:02.868966 #26010] [] DEBUG -- DirectorJobRunner: (0.000269s) (conn: 47388435421440) INSERT INTO "events" ("parent_id", "timestamp", "user", "action", "object_type", "object_name", "error", "task", "deployment", "instance", "context_json") VALUES (NULL, '2019-02-07 22:25:02.866764+0000', 'admin', 'release', 'lock', 'lock:release:cflinuxfs3', NULL, '148', NULL, NULL, '{}') RETURNING *
D, [2019-02-07T22:25:02.869566 #26010] [] DEBUG -- DirectorJobRunner: (0.003657s) (conn: 47388448342640) COMMIT
D, [2019-02-07T22:25:02.871028 #26010] [] DEBUG -- DirectorJobRunner: (0.001627s) (conn: 47388435421440) COMMIT
D, [2019-02-07T22:25:02.871266 #26010] [] DEBUG -- DirectorJobRunner: Thread is no longer needed, cleaning up
D, [2019-02-07T22:25:02.871750 #26010] [] DEBUG -- DirectorJobRunner: (0.000042s) (conn: 47388435164720) BEGIN
D, [2019-02-07T22:25:02.872609 #26010] [] DEBUG -- DirectorJobRunner: (0.000502s) (conn: 47388435164720) UPDATE "tasks" SET "event_output" = ("event_output" || '{"time":1549578302,"stage":"Deleting releases","tags":[],"total":4,"task":"loggregator-agent/3.4","index":4,"state":"failed","progress":100,"data":{"error":"PG::ForeignKeyViolation: ERROR:  update or delete on table \"templates\" violates foreign key constraint \"instances_templates_template_id_fkey\" on table \"instances_templates\"\nDETAIL:  Key (id)=(139) is still referenced from table \"instances_templates\".\n"}}
') WHERE ("id" = 148)
D, [2019-02-07T22:25:02.879580 #26010] [] DEBUG -- DirectorJobRunner: (0.006811s) (conn: 47388435164720) COMMIT
D, [2019-02-07T22:25:02.883082 #26010] [] DEBUG -- DirectorJobRunner: (0.001585s) (conn: 47388435421440) DELETE FROM "locks" WHERE (("name" = 'lock:release:loggregator-agent') AND ("uid" = '0ba45091-7b2f-4909-af0b-506c2249877c'))
D, [2019-02-07T22:25:02.883195 #26010] [] DEBUG -- DirectorJobRunner: Deleted lock: lock:release:loggregator-agent uid: 0ba45091-7b2f-4909-af0b-506c2249877c
D, [2019-02-07T22:25:02.884262 #26010] [] DEBUG -- DirectorJobRunner: (0.000046s) (conn: 47388435164720) BEGIN
D, [2019-02-07T22:25:02.885099 #26010] [] DEBUG -- DirectorJobRunner: (0.000271s) (conn: 47388435164720) INSERT INTO "events" ("parent_id", "timestamp", "user", "action", "object_type", "object_name", "error", "task", "deployment", "instance", "context_json") VALUES (NULL, '2019-02-07 22:25:02.883421+0000', 'admin', 'release', 'lock', 'lock:release:loggregator-agent', NULL, '148', NULL, NULL, '{}') RETURNING *
D, [2019-02-07T22:25:02.886719 #26010] [] DEBUG -- DirectorJobRunner: (0.001116s) (conn: 47388435164720) COMMIT
E, [2019-02-07T22:25:02.887686 #26010] [] ERROR -- DirectorJobRunner: Worker thread raised exception: PG::ForeignKeyViolation: ERROR:  update or delete on table "templates" violates foreign key constraint "instances_templates_template_id_fkey" on table "instances_templates"
DETAIL:  Key (id)=(139) is still referenced from table "instances_templates".
 - /var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/adapters/postgres.rb:166:in `async_exec'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/adapters/postgres.rb:166:in `block in execute_query'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/database/logging.rb:49:in `log_connection_yield'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/adapters/postgres.rb:166:in `execute_query'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/adapters/postgres.rb:153:in `block in execute'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/adapters/postgres.rb:129:in `check_disconnect_errors'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/adapters/postgres.rb:153:in `execute'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/adapters/postgres.rb:515:in `_execute'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/adapters/postgres.rb:327:in `block (2 levels) in execute'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/adapters/postgres.rb:537:in `check_database_errors'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/adapters/postgres.rb:327:in `block in execute'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/database/connecting.rb:301:in `block in synchronize'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/connection_pool/threaded.rb:103:in `hold'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/database/connecting.rb:301:in `synchronize'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/adapters/postgres.rb:327:in `execute'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/database/query.rb:54:in `execute_dui'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/dataset/actions.rb:1146:in `execute_dui'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/dataset/actions.rb:977:in `with_sql_delete'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/model/base.rb:2093:in `_delete_without_checking'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/model/base.rb:2075:in `_delete'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/model/base.rb:1528:in `delete'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/model/base.rb:2131:in `_destroy_delete'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/model/base.rb:2115:in `block in _destroy'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/model/base.rb:1382:in `around_destroy'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/model/base.rb:2109:in `_destroy'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/model/base.rb:1543:in `block (2 levels) in destroy'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/model/base.rb:2372:in `block in checked_transaction'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/database/transactions.rb:224:in `_transaction'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/database/transactions.rb:199:in `block in transaction'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/database/connecting.rb:301:in `block in synchronize'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/connection_pool/threaded.rb:107:in `hold'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/database/connecting.rb:301:in `synchronize'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/database/transactions.rb:165:in `transaction'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/model/base.rb:2372:in `checked_transaction'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/model/base.rb:1543:in `block in destroy'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/model/base.rb:2360:in `checked_save_failure'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/sequel-4.49.0/lib/sequel/model/base.rb:1543:in `destroy'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/helpers/template_deleter.rb:19:in `delete'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/helpers/release_version_deleter.rb:79:in `block (2 levels) in delete_release_version'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/helpers/release_version_deleter.rb:103:in `block in track_and_log'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/event_log.rb:105:in `advance_and_track'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/helpers/release_version_deleter.rb:101:in `track_and_log'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/helpers/release_version_deleter.rb:75:in `block in delete_release_version'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/helpers/release_version_deleter.rb:74:in `each'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/helpers/release_version_deleter.rb:74:in `delete_release_version'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/helpers/release_version_deleter.rb:25:in `delete'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/helpers/name_version_release_deleter.rb:44:in `delete_release_version'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/helpers/name_version_release_deleter.rb:17:in `find_and_delete_release'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/cleanup_artifacts.rb:74:in `block (6 levels) in perform'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/event_log.rb:105:in `advance_and_track'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/cleanup_artifacts.rb:73:in `block (5 levels) in perform'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/cleanup_artifacts.rb:72:in `each'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/cleanup_artifacts.rb:72:in `block (4 levels) in perform'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/lock_helper.rb:30:in `block in with_release_lock'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/lock_helper.rb:42:in `with_release_locks'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/lock_helper.rb:30:in `with_release_lock'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/jobs/cleanup_artifacts.rb:71:in `block (3 levels) in perform'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh_common-0.0.0/lib/common/thread_pool.rb:77:in `block (2 levels) in create_thread'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh_common-0.0.0/lib/common/thread_pool.rb:63:in `loop'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/bosh_common-0.0.0/lib/common/thread_pool.rb:63:in `block in create_thread'
/var/vcap/data/packages/director/0f4b5b37e1a6b056dcb41e749d459968bd35e7e6/gem_home/ruby/2.4.0/gems/logging-2.2.2/lib/logging/diagnostic_context.rb:474:in `block in create_with_logging_context'
D, [2019-02-07T22:25:02.887793 #26010] [] DEBUG -- DirectorJobRunner: Thread is no longer needed, cleaning up
D, [2019-02-07T22:25:02.887878 #26010] [task:148] DEBUG -- DirectorJobRunner: Shutting down pool
```

**Versions (please complete the following information):**
 - Infrastructure: `GCP`
 - BOSH version `268.3.0`
 - BOSH CLI version `5.4.0-891ff634-2018-11-14T00:21:14Z`
 - Stemcell version `ubuntu-xenial/250.4`

**Deployment info:**
We used bbl to deploy our director, all the commands necessary to reproduce the issue are aforementioned.

**Additional context**
Tracker story: https://www.pivotaltracker.com/story/show/163771856
Slack conversation: https://cloudfoundry.slack.com/messages/C02HPPYQ2/
Build failure: https://release-integration.ci.cf-app.com/teams/main/pipelines/cf-deployment/jobs/experimental-deploy/builds/1069

**Comment by dpb587-pivotal** [comment]
I think this is a duplicate of https://github.com/cloudfoundry/bosh/issues/2069 which appears to be fixed in [v268.4.0](https://github.com/cloudfoundry/bosh/releases/tag/v268.4.0). Are you able to verify the fix with a newer director?

**Comment by cdutra** [comment]
We tested on director 268.5.0 and it's fixed, thanks!
